go/multiorch/recovery/engine.go

```diff
@@ -216,6 +266,66 @@ func (re *Engine) initMetrics() {
 	if err != nil {
 		re.logger.Error("failed to create refresh_latency histogram", "error", err)
 	}
+
+	// Counter for health check attempts
+	re.pollAttemptsCounter, err = meter.Int64Counter(
+		"multiorch.recovery.poll_attempts",
+		metric.WithDescription("Total number of pooler health check attempts"),
+		metric.WithUnit("{checks}"),
+	)
+	if err != nil {
+		re.logger.Error("failed to create poll_attempts counter", "error", err)
+	}
+
+	// Counter for health check failures
+	re.pollFailuresCounter, err = meter.Int64Counter(
+		"multiorch.recovery.poll_failures",
+		metric.WithDescription("Total number of pooler health check failures"),
+		metric.WithUnit("{failures}"),
+	)
+	if err != nil {
+		re.logger.Error("failed to create poll_failures counter", "error", err)
+	}
+
+	// Counter for polls exceeding interval
+	re.poolerPollExceededCounter, err = meter.Int64Counter(
+		"multiorch.recovery.poll_interval_exceeded",
+		metric.WithDescription("Number of health checks that exceeded the poll interval"),
+		metric.WithUnit("{checks}"),
+	)
+	if err != nil {
+		re.logger.Error("failed to create poll_interval_exceeded counter", "error", err)
+	}
+
+	// Histogram for individual poll latency
+	re.pollLatency, err = meter.Float64Histogram(
+		"multiorch.recovery.poll_latency",
+		metric.WithDescription("Latency of individual pooler health checks"),
+		metric.WithUnit("s"),
+	)
+	if err != nil {
+		re.logger.Error("failed to create poll_latency histogram", "error", err)
+	}
+
+	// Histogram for health check cycle duration
+	re.healthCheckCycleDuration, err = meter.Float64Histogram(
+		"multiorch.recovery.health_check_cycle_duration",
+		metric.WithDescription("Duration of complete health check cycles"),
+		metric.WithUnit("s"),
+	)
+	if err != nil {
+		re.logger.Error("failed to create health_check_cycle_duration histogram", "error", err)
+	}
+
+	// Histogram for poolers checked per cycle
+	re.poolersCheckedPerCycle, err = meter.Int64Histogram(
```

Comment:
I'm not sure how well otel's guesses at histogram buckets will work for this.

What's kind of situations are you thinking will be debugged with this histogram?